libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1410,6 +1410,21 @@ bool TypeChecker::visit(FunctionCall const& _functionCall)
 	else
 		_functionCall.annotation().type = make_shared<TupleType>(functionType->returnParameterTypes());
 
+	// Internal library functions must be implemented, because of inlining rules.
+	if (
+		functionType->kind() == FunctionType::Kind::Internal &&
+		functionType->hasDeclaration() &&
+		dynamic_cast<FunctionDefinition const*>(&functionType->declaration())
+	)
+	{
+		FunctionDefinition const* function = dynamic_cast<FunctionDefinition const*>(&functionType->declaration());
+		bool isLibraryFunction =
+			dynamic_cast<ContractDefinition const*>(function->scope()) &&
+			dynamic_cast<ContractDefinition const*>(function->scope())->isLibrary();
+		if (!function->isImplemented() && isLibraryFunction)
```

Comment:
Ah, now I see the test case. Perhaps we  should rather check whether the current contract is implemented instead of whether the called function is a library function?